src/ui/find-ui/tests/FamilyHubs.ServiceDirectory.UnitTests/Web/BaseWebTest.cs

```diff
@@ -0,0 +1,111 @@
+    private readonly HttpClient _client;
+    private readonly WebApplicationFactory<Program> _factory;
+
+    private class MyWebApplicationFactory : WebApplicationFactory<Program>
+    {
+        protected override IHost CreateHost(IHostBuilder builder)
+        {
+            builder.ConfigureHostConfiguration(config =>
+            {
+                config
+                    .AddInMemoryCollection(new Dictionary<string, string?>
+                        {
+                            { "GovUkOidcConfiguration:StubAuthentication:UseStubAuthentication", "true" },
+                            { "GovUkOidcConfiguration:StubAuthentication:UseStubClaims", "true" },
+                            { "GovUkOidcConfiguration:Oidc:BaseUrl", "https://example.com" }
+                        }
+                    );
+            });
+
+            return base.CreateHost(builder);
+        }
+    }
+
+    protected BaseWebTest()
+    {
+        _factory = new MyWebApplicationFactory();
+        _client = _factory.WithWebHostBuilder(builder =>
+            {
+                builder.ConfigureTestServices(Configure);
+            }
+        ).CreateClient(
+            new WebApplicationFactoryClientOptions {
+                BaseAddress = new Uri(BaseUrl),
+                HandleCookies = true
+            }
+        );
+    }
+
+    protected virtual void Configure(IServiceCollection services)
+    {
+        
+    }
+
+    protected async Task<IHtmlDocument> Navigate(string uri, Action<HttpResponseMessage>? responseValidation = null)
+    {
+        var response = await _client.GetAsync(uri);
+        responseValidation?.Invoke(response);
+        return await GetDocumentAsync(response);
+    }
+
+    private static async Task<IHtmlDocument> GetDocumentAsync(HttpResponseMessage response)
+    {
+        var content = await response.Content.ReadAsStringAsync();
+        var document = await BrowsingContext.New(Configuration.Default.WithCss()).OpenAsync(ResponseFactory, CancellationToken.None);
+        return (IHtmlDocument) document;
+
+        void ResponseFactory(VirtualResponse htmlResponse)
+        {
+            htmlResponse
+                .Address(response.RequestMessage!.RequestUri)
+                .Status(response.StatusCode);
+
+            MapHeaders(response.Headers);
+            MapHeaders(response.Content.Headers);
+
+            htmlResponse.Content(content);
+
+            void MapHeaders(HttpHeaders headers)
+            {
+                foreach (var header in headers)
+                {
+                    foreach (var value in header.Value)
+                    {
+                        htmlResponse.Header(header.Key, value);
+                    }
+                }
+            }
+        }
+    }
+
+    protected virtual void Dispose(bool disposing)
+    {
+        // Cleanup
+    }
+
+    public void Dispose()
+    {
+        Dispose(true);
+
+        _client.Dispose();
```

Comment:
Following the IDisposable pattern, shouldn't the _client.Dispose() and _factory.Dispose() go in the virtual Dispose(bool disposing) method and call dispose on those objects when disposing id true?

Reply:
They should, this was copied from an existing file in the project.
Turns out this is implemented wrong in this repo a lot and I've fixed some other instances in [my other testing ticket](https://github.com/DFE-Digital/fh-services/pull/226)